docs/content/guide/component.ngdoc

```diff
@@ -282,7 +282,7 @@ it upwards to the heroList component, which updates the original data.
 </file>
 <file name="editableField.html">
   <span ng-switch="$ctrl.editMode">
-    <input ng-switch-when="true" type="text" ng-model="$ctrl.fieldValue">
+    <input ng-switch-when="true" type="{{::$ctrl.fieldType}}" ng-model="$ctrl.fieldValue">
```

Comment:
> Hmm, is not it the best practice for performance?

It is, but only when the usecase allows it. This is avery simple demo, of course, but in a real up, having a very generic, reusable `editableField` you'd probably want to allow consumers to change the field type dynamically.

In any case, the main purpose of the is to showcase components and their usage and related use-patterns and to that end I believe it's best to keep other concepts to a minimum. I wouldn't oppose if `::` was aready used elsewhere (e.g. I'm sure there are places where is this guide where it is much more appropriate - but this is not the point), but introducing it here unnecessarily introduces yet another concept.

So, I would prefer it gone (but it's not super important, so feel free to leave it as is if you feel strongly about it :smiley:)